easy_infra/utils.py

```diff
@@ -258,6 +286,19 @@ def gather_tools_and_environments(
     return image_and_tool_and_environment_tags
 
 
+def gather_users(*, user: str) -> list[str]:
+    """Return a list of users, based on the simplified provided user"""
+    if user == "all":
+        users: list[str] = constants.USERS
+    elif user not in constants.USERS:
+        LOG.error(f"{user} is not a supported user, exiting...")
+        sys.exit(1)
+    else:
+        users: list[str] = [user]
+
+    return users
```

Comment:
Wouldnt it be better to fail immediately if the provided user does not exist?
```suggestion
def gather_users(*, user: str) -> list[str]:
    """Return a list of users, based on the simplified provided user"""
    if user not in constants.USERS:
         LOG.error(f"{user} is not a supported user, exiting...")
         sys.exit(1)
     elif user == "all":
           users: list[str] = constants.USERS
      else:
            users: list[str] = [user]
```